duckduckgo/iOS
# Downloads list + enhanced file handling

Task/Issue URL: https://app.asana.com/0/414709148257752/1145004832722773/f
Tech Design URL: https://app.asana.com/0/481882893211075/1201601635648875 , https://app.asana.com/0/481882893211075/1201600067271018


**Description**:
Provide comprehensive file handling support in iOS browser to enable:
- **Opening files rendered within a tab** (common file type e.g. PDF) and **sharing/exporting them to other apps** (e.g. Books app, Mail or Messages etc.)
- **Opening specific file types that can be previewed and acted on in place** (without enqueuing the download on the list) - currently supported for **Apple Wallet passes** and **AR files**
- **Downloading other file types** (_whenever file is not covered by above cases_ OR _when a know file type is served with `application/octet-stream` MIME type_).  Download is started and the completed file is available on the **downloads list** where it can be shared/exported to other more suitable app.

**Steps to test this PR**:
Please refer the test download links found here: https://app.asana.com/0/0/1201526490030383/f
1. Common file types should open directly within tab (e.g. PDF documents, images, iWork and Office documents)
2. Files opened within the tab should be properly exported vis "Share" button to external apps (e.g. PDF file -> Books app)
3. Preview for Apple Wallet passes should open in separate modal view.
4. AR files should open in separate modal view.
5. Other files (or files served with `application/octet-stream` MIME type) should open alert asking to download.
6. Downloaded files should be added to "Downloads list" (available from the menu) where they can be previewed, shared and managed.


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:
